gherkin/go/go.mod

```diff
@@ -1,7 +1,7 @@
 module github.com/cucumber/gherkin-go/v19
 
 require (
-	github.com/cucumber/messages-go/v16 v16.0.1
+	github.com/cucumber/common/messages/v16 v16.0.0
```

Comment:
There is no Git tag named `v16.0.0` in the monorepo. all tags are of the form `library/version`, so you need to use something like this:

```
github.com/cucumber/common/messages/v16 messages/v16.0.0
```

I haven't tested if this is valid go module syntax. If it isn't, we have to start using a different naming scheme for tags - perhaps with a hyphen or dot instead of a slash,

Reply:
It seems to have ways of inferring the tag from the version string that I can't quite fathom yet. If you see the first error (`...at revision messages/v16.0.0`) it seems to have figured out the tag name and found it, but it has the wrong path to the actual `go.mod` file (`missing github.com/cucumber/common/messages/go.mod`)

Reply:
In the second case it makes a different guess for the tag name, and can't find it in git: `unknown revision messages/go/v16.0.0`

Reply:
go look for the module on the master branch.
But it does not exist (yet) because messages/go.mod still have `module github.com/cucumber/messages-go/v16`

with the following in `gherkin/go/go.mod` I was able to have a coherent error message:
```
github.com/cucumber/common/messages/go/v16 e0d826edaa5d24e50910eaa689194a8b3340c907

# go: github.com/cucumber/common/messages/go/v16@v16.0.0-20210611214814-e0d826edaa5d: parsing go.mod:
#         module declares its path as: github.com/cucumber/common/messages/v16
#                 but was required as: github.com/cucumber/common/messages/go/v16
```

I think we will have to update the modules like the following (in `messages/go/go.mod`):
```
module github.com/cucumber/common/messages/go/v16
```

So we may need at least 2 PR to make it work:

The first one updating all go.mod replacing the module path from the subrepos to the appropriate monorepo sub-directory (for example `module github.com/cucumber/messages-go/v16` => `module github.com/cucumber/common/messages/go/v16`)

Then a second PR to update all the go `require` directive to point to the new location.